lib/ansible/utils/display.py

```diff
@@ -199,6 +224,13 @@ def __init__(self, verbosity=0):
 
         self._set_column_width()
 
+        try:
+            # NB: we're relying on the display singleton behavior to ensure this only runs once
+            _synchronize_textiowrapper(sys.stdout, self._lock)
+            _synchronize_textiowrapper(sys.stderr, self._lock)
+        except ValueError:
+            self.warning("failed to patch stdout/stderr for fork-safety")
```

Comment:
I take it that this is the reason why we are leaving the `with self._lock:` in `Display.display`?  In the event that we cannot patch stdout/stderr?

Reply:
Correct- since it's an RLock and should stay that way, the incremental cost when we already own the lock is pretty minimal, and still provides the safety around intermingled output in situations where we couldn't take the lock.